app/server/models/credit_transfer.py

```diff
@@ -316,9 +316,8 @@ def resolve_as_complete(self, batch_uuid=None):
         if (datetime.datetime.utcnow() - self.created).seconds > 5:
             clear_metrics_cache()
             rebuild_metrics_cache()
-        if self.transfer_type == TransferTypeEnum.PAYMENT and self.transfer_subtype == TransferSubTypeEnum.DISBURSEMENT:
-            if self.recipient_user and self.recipient_user.transfer_card:
-                self.recipient_user.transfer_card.update_transfer_card()
```

Comment:
It seems like the original code, only updated on `DISBURSEMENT`, so a very old bug from how the system was originally designed.

https://github.com/teamsempo/SempoBlockchain/blame/e0d784762681ebfbf039bdc7a47a88cc7f5a53d5/app/server/models/credit_transfer.py#L169-L171